### PR TITLE
Adding LTR for MI to .NET SDK

### PR DESCRIPTION
**Pull request by xaliciayang**
Adding clients + test file for Managed Instance Long Term Retention Tests

### CODE REVIEW

**Review by isra-fel — CHANGES_REQUESTED** [review]
Hi @xaliciayang , it failed to build because some of the models could not be found, e.g. `ManagedInstanceLongTermRetentionBackup`. Did you forget to check in them?

Besides, please
- add link(s) to the swagger spec review in this PR's description
- check in the metadata txt of your module under `/eng/mgmt/mgmtmetadata` 

Detailed process guide: https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#generating-net-sdk

**Comment by xaliciayang** [comment]
Added the missing models (also added many unrelated files in attempt to sync local repo). Will remove out-of-scope files

Here is the swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/8266

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/IManagedDatabasesOperations.cs [hunk 1]
```diff
@@ -174,6 +174,33 @@ public partial interface IManagedDatabasesOperations
...
+        /// <exception cref="Microsoft.Rest.Azure.CloudException">
+        /// Thrown when the operation returned an invalid status code
+        /// </exception>
+        /// <exception cref="Microsoft.Rest.SerializationException">
+        /// Thrown when unable to deserialize the response
+        /// </exception>
+        /// <exception cref="Microsoft.Rest.ValidationException">
+        /// Thrown when a required parameter is null
+        /// </exception>
+        Task<AzureOperationResponse<IPage<ManagedDatabase>>> ListInaccessibleByInstanceWithHttpMessagesAsync(string resourceGroupName, string managedInstanceName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (pixia) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/IManagedDatabasesOperations.cs]: this is not our change

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/IManagedDatabasesOperations.cs [hunk 2]
```diff
@@ -348,5 +375,27 @@ public partial interface IManagedDatabasesOperations
...
+        /// <exception cref="Microsoft.Rest.Azure.CloudException">
+        /// Thrown when the operation returned an invalid status code
+        /// </exception>
+        /// <exception cref="Microsoft.Rest.SerializationException">
+        /// Thrown when unable to deserialize the response
+        /// </exception>
+        /// <exception cref="Microsoft.Rest.ValidationException">
+        /// Thrown when a required parameter is null
+        /// </exception>
+        Task<AzureOperationResponse<IPage<ManagedDatabase>>> ListInaccessibleByInstanceNextWithHttpMessagesAsync(string nextPageLink, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (pixia) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/IManagedDatabasesOperations.cs]: not our change

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/SessionRecords/ManagedInstanceLongTermRetentionTests/TestManagedInstanceLongTermRetentionPolicies.json [hunk 3]
```diff
@@ -0,0 +1,828 @@
...
+      },
+      "ResponseHeaders": {
+        "Cache-Control": [
+          "no-cache"
+        ],
+        "Pragma": [
+          "no-cache"
+        ],
+        "Location": [
+          "https://api-dogfood.resources.windows-int.net/subscriptions/8cfb8b62-bcd6-4713-89ad-18097f75cc5b/providers/Microsoft.Sql/locations/southeastasia/managedDatabaseOperationResults/1dd83660-8f4f-4079-952d-2fb139812065?api-version=2019-06-01-preview"
```
Comment (pixia) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/SessionRecords/ManagedInstanceLongTermRetentionTests/TestManagedInstanceLongTermRetentionPolicies.json]: are we ok to expose test endpoint?

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/Models/DatabaseState6.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+// regenerated.
+// </auto-generated>
+
+namespace Microsoft.Azure.Management.Sql.Models
+{
+
+    /// <summary>
+    /// Defines values for DatabaseState6.
+    /// </summary>
+    public static class DatabaseState6
```
Comment (jaredmoo) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/Models/DatabaseState6.cs]: Autorest bug??

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SdkInfo_SqlManagementClient.cs [hunk 5]
```diff
@@ -87,7 +89,7 @@ internal static partial class SdkInfo
                 new Tuple<string, string, string>("Sql", "RestorePoints", "2017-03-01-preview"),
                 new Tuple<string, string, string>("Sql", "SensitivityLabels", "2017-03-01-preview"),
                 new Tuple<string, string, string>("Sql", "ServerAutomaticTuning", "2017-03-01-preview"),
-                new Tuple<string, string, string>("Sql", "ServerAzureADAdministrators", "2014-04-01"),
```
Comment (xaliciayang) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SdkInfo_SqlManagementClient.cs]: @amolagar should I revert this and leave it to you?

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SqlManagementClient.cs [hunk 6]
```diff
@@ -610,33 +607,6 @@ public SqlManagementClient(ServiceClientCredentials credentials, params Delegati
...
-        /// </param>
-        /// <param name='httpClient'>
-        /// HttpClient to be used
-        /// </param>
-        /// <param name='disposeHttpClient'>
-        /// True: will dispose the provided httpClient on calling SqlManagementClient.Dispose(). False: will not dispose provided httpClient</param>
-        /// <exception cref="System.ArgumentNullException">
-        /// Thrown when a required parameter is null
-        /// </exception>
-        public SqlManagementClient(ServiceClientCredentials credentials, HttpClient httpClient, bool disposeHttpClient) : this(httpClient, disposeHttpClient)
```
Comment (pixia) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SqlManagementClient.cs]: why is this removed?

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SqlManagementClient.cs [hunk 7]
```diff
@@ -483,37 +468,49 @@ public partial class SqlManagementClient : ServiceClient<SqlManagementClient>, I
...
         /// <summary>
-        /// Initializes a new instance of the SqlManagementClient class.
+        /// Gets the ILongTermRetentionManagedInstanceBackupsOperations.
         /// </summary>
-        /// <param name='httpClient'>
-        /// HttpClient to be used
-        /// </param>
-        /// <param name='disposeHttpClient'>
-        /// True: will dispose the provided httpClient on calling SqlManagementClient.Dispose(). False: will not dispose provided httpClient</param>
-        protected SqlManagementClient(HttpClient httpClient, bool disposeHttpClient) : base(httpClient, disposeHttpClient)
```
Comment (pixia) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SqlManagementClient.cs]: why is this removed?

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SdkInfo_SqlManagementClient.cs [hunk 8]
```diff
@@ -110,19 +112,10 @@ internal static partial class SdkInfo
                 new Tuple<string, string, string>("Sql", "Usages", "2018-06-01-preview"),
                 new Tuple<string, string, string>("Sql", "VirtualClusters", "2015-05-01-preview"),
                 new Tuple<string, string, string>("Sql", "VirtualNetworkRules", "2015-05-01-preview"),
+                new Tuple<string, string, string>("Sql", "WorkloadClassifiers", "2019-06-01-preview"),
```
Comment (pixia) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SdkInfo_SqlManagementClient.cs]: remove workloadClassifiers

**Review by pixia — APPROVED** [review]

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj [hunk 9]
```diff
@@ -12,9 +12,11 @@
 		<PackageReleaseNotes>
 			<![CDATA[
 New features:
-- Added support for Azure SQL Server Azure Active Directory administrator API version 2019-06-01-preview.
+- Added support for Azure SQL Server Azure Active Directory administrator API` version 2019-06-01-preview.
```
Comment (isra-fel) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj]: nit:
```suggestion
- Added support for Azure SQL Server Azure Active Directory administrator API version 2019-06-01-preview.
```

**Review by isra-fel — APPROVED** [review]
